File: CustomDataGridColumns/Operations.vb
Public Class Operations
    Private Shared ConnectionString As String =
                       "Data Source=.\SQLEXPRESS;" &
                       "Initial Catalog=NorthWind2020;Integrated Security=True"

    Public Shared Function Read() As DataTable
        Using cn As New SqlClient.SqlConnection With {.ConnectionString = ConnectionString}
            Using cmd As New SqlClient.SqlCommand With {.Connection = cn}


                cmd.CommandText =
                    <SQL>
                        SELECT C.CompanyName, E.FirstName + ' ' + E.LastName AS fullname, O.OrderDate 
                        FROM Orders AS O 
                        INNER JOIN Customers AS C ON O.CustomerIdentifier = C.CustomerIdentifier 
                        INNER JOIN Employees AS E ON O.EmployeeID = E.EmployeeID;
                    </SQL>.Value
                Dim orderTable As New DataTable

                cn.Open()

                orderTable.Load(cmd.ExecuteReader())

                Return orderTable

            End Using
        End Using
    End Function
End Class

File: WorkingWithAccess/Classes/Operations.cs
using System.Data;
using System.Data.OleDb;

namespace WorkingWithAccess.Classes;

public class Operations
{

    public static string ConnectionString = 
        "Provider=Microsoft.ACE.OLEDB.12.0;Data Source=Database1.accdb";

    public static (bool success, DataTable table) CanConnect()
    {
        DataTable categoryTable = new DataTable();
        try
        {
            
            using OleDbConnection cn = new() { ConnectionString = ConnectionString };
            using OleDbCommand cmd = new() { Connection = cn };

            cmd.CommandText = "SELECT Identifier, Category FROM  Category ORDER BY Category";


            cn.Open();

            categoryTable.Load(cmd.ExecuteReader());

            return (true,categoryTable);
        }
        catch 
        {
            return (false, categoryTable);
        }
    }

}